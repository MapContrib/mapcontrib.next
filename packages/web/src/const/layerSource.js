export const OSMOSE_SOURCE = 'OSMOSE_SOURCE';
export const OVERPASS_SOURCE = 'OVERPASS_SOURCE';
