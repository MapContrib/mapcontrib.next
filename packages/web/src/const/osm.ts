export const CHANGESET_CREATED_BY = 'MapContrib {version}';
export const CHANGESET_COMMENT =
  'Contribution sent from the following MapContrib theme: {url}';
