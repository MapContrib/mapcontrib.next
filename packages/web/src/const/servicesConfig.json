{
  "OSMOSE_PROD_URL": "https://opendata.osmose.openstreetmap.fr/fr/api/0.2",
  "OVERPASS_PROD_URL": "https://overpass-api.de/api",

  "OSM_PROD_URL": "https://www.openstreetmap.org",
  "OSM_DEV_URL": "https://api06.dev.openstreetmap.org",

  "OAUTH_SECRET": "aaQcLOJFno7EGM0hC8nXaE85zOY4UP9uxklqpo7O",
  "OAUTH_CONSUMER_KEY": "WiLOF1hp5Aloyldyd4VLN2ZjBvqoyUNO3n2mDCpL"
}
