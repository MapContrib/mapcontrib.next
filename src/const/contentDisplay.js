export const CONTENT_DISPLAY_MODAL = 'CONTENT_DISPLAY_MODAL';
export const CONTENT_DISPLAY_SIDEBAR = 'CONTENT_DISPLAY_SIDEBAR';
