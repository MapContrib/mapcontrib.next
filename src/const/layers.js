export const sourceTypes = Object.freeze({
  OSMOSE: 'OSMOSE',
  OVERPASS: 'OVERPASS'
});
