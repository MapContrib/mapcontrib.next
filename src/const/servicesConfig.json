{
  "OSMOSE_URL": "https://opendata.osmose.openstreetmap.fr/fr/api/0.2"
}
