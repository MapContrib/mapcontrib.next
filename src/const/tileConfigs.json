[
    {
        "id": "osm",
        "tileSources": ["osm"]
    },
    {
        "id": "mapboxStreetsSatellite",
        "tileSources": ["mapboxStreetsSatellite"]
    },
    {
        "id": "watercolor",
        "tileSources": ["watercolor"]
    },
    {
        "id": "osmMonochrome",
        "tileSources": ["osmMonochrome"]
    },
    {
        "id": "osmOutdoors",
        "tileSources": ["osmOutdoors"]
    },
    {
        "id": "osmRoads",
        "tileSources": ["osmRoads"]
    },
    {
        "id": "landscape",
        "tileSources": ["landscape"]
    },
    {
        "id": "toner",
        "tileSources": ["toner"]
    },
    {
        "id": "transport",
        "tileSources": ["transport"]
    },
    {
        "id": "hot",
        "tileSources": ["hot"]
    },
    {
        "id": "openCycleMap",
        "tileSources": ["openCycleMap"]
    },
    {
        "id": "hikeBikeMap",
        "tileSources": ["hikeBikeMap"]
    },
    {
        "id": "hydda",
        "tileSources": ["hydda"]
    },
    {
        "id": "openTopoMap",
        "tileSources": ["openTopoMap"]
    },
    {
        "id": "openRiverboatMap",
        "tileSources": ["openRiverboatMap"]
    },
    {
        "id": "osmFr",
        "tileSources": ["osmFr"]
    },
    {
        "id": "osmFrBano",
        "tileSources": ["osmFr", "bano"]
    },
    {
        "id": "cadastre",
        "tileSources": ["cadastre"]
    },
    {
        "id": "osmDe",
        "tileSources": ["osmDe"]
    }
]
