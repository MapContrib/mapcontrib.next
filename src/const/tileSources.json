[
    {
        "id": "osm",
        "name": "OpenStreetMap",
        "attribution": "Data &copy; <a href=\"http://www.openstreetmap.org/copyright\" rel=\"noopener noreferrer\" target=\"_blank\">OpenStreetMap</a>",
        "urlTemplate": "https://{s}.tile.openstreetmap.org/{z}/{x}/{y}.png",
        "minZoom": 0,
        "maxZoom": 19
    },
    {
        "id": "mapboxStreetsSatellite",
        "name": "Mapbox Streets Satellite",
        "attribution": "Données &copy; <a href=\"http://www.openstreetmap.org/copyright\" rel=\"noopener noreferrer\" target=\"_blank\">OpenStreetMap</a> - Tiles <a href=\"https://www.mapbox.com\" rel=\"noopener noreferrer\" target=\"_blank\">Mapbox</a>",
        "urlTemplate": "https://{s}.tiles.mapbox.com/v4/mapbox.streets-satellite/{z}/{x}/{y}.png?access_token=pk.eyJ1IjoibWFwY29udHJpYiIsImEiOiJjaWt6YTd1OTYwMDY3d25tMHN2b2hhaTJuIn0.HtC_5kFI2ZEZs7ouqfXTcw",
        "minZoom": 0,
        "maxZoom": 22
    },
    {
        "id": "watercolor",
        "name": "Watercolor",
        "attribution": "Tiles <a href=\"http://stamen.com\" rel=\"noopener noreferrer\" target=\"_blank\">Stamen Design</a> under <a href=\"http://creativecommons.org/licenses/by/3.0\" rel=\"noopener noreferrer\" target=\"_blank\">CC BY 3.0</a> - Data &copy; <a href=\"http://www.openstreetmap.org/copyright\" rel=\"noopener noreferrer\" target=\"_blank\">OpenStreetMap</a>",
        "urlTemplate": "http://{s}.tile.stamen.com/watercolor/{z}/{x}/{y}.jpg",
        "minZoom": 0,
        "maxZoom": 18
    },
    {
        "id": "osmMonochrome",
        "name": "OSM Monochrome",
        "attribution": "Data &copy; <a href=\"http://www.openstreetmap.org/copyright\" rel=\"noopener noreferrer\" target=\"_blank\">OpenStreetMap</a>",
        "urlTemplate": "http://{s}.tiles.wmflabs.org/bw-mapnik/{z}/{x}/{y}.png",
        "minZoom": 1,
        "maxZoom": 18
    },
    {
        "id": "osmOutdoors",
        "name": "OSM Outdoors",
        "attribution": "Tiles &copy; <a href=\"http://www.thunderforest.com/outdoors\" rel=\"noopener noreferrer\" target=\"_blank\">Gravitystorm</a> - Data &copy; <a href=\"http://www.openstreetmap.org/copyright\" rel=\"noopener noreferrer\" target=\"_blank\">OpenStreetMap</a>",
        "urlTemplate": "https://{s}.tile.thunderforest.com/outdoors/{z}/{x}/{y}.png",
        "minZoom": 0,
        "maxZoom": 20
    },
    {
        "id": "osmRoads",
        "name": "OSM Roads",
        "attribution": "Tiles &copy; <a href=\"http://giscience.uni-hd.de\" rel=\"noopener noreferrer\" target=\"_blank\">GIScience Research Group at Heidelberg University</a> - Data &copy;  <a href=\"http://www.openstreetmap.org/copyright\" rel=\"noopener noreferrer\" target=\"_blank\">OpenStreetMap</a>",
        "urlTemplate": "http://korona.geog.uni-heidelberg.de/tiles/roads/x={x}&y={y}&z={z}",
        "minZoom": 0,
        "maxZoom": 19
    },
    {
        "id": "landscape",
        "name": "Landscape",
        "attribution": "Tiles &copy; <a href=\"http://www.thunderforest.com/outdoors\" rel=\"noopener noreferrer\" target=\"_blank\">Gravitystorm</a> - Data <a href=\"http://www.openstreetmap.org/copyright\" rel=\"noopener noreferrer\" target=\"_blank\">OpenStreetMap</a>",
        "urlTemplate": "http://{s}.tile3.opencyclemap.org/landscape/{z}/{x}/{y}.png",
        "minZoom": 0,
        "maxZoom": 20
    },
    {
        "id": "toner",
        "name": "Toner",
        "attribution": "Tiles <a href=\"http://stamen.com\" rel=\"noopener noreferrer\" target=\"_blank\">Stamen Design</a> under <a href=\"http://creativecommons.org/licenses/by/3.0\" rel=\"noopener noreferrer\" target=\"_blank\">CC BY 3.0</a> - Data &copy; <a href=\"http://www.openstreetmap.org/copyright\" rel=\"noopener noreferrer\" target=\"_blank\">OpenStreetMap</a>",
        "urlTemplate": "http://{s}.tile.stamen.com/toner-lite/{z}/{x}/{y}.png",
        "minZoom": 0,
        "maxZoom": 20
    },
    {
        "id": "transport",
        "name": "Transport",
        "attribution": "Tiles &copy; <a href=\"http://www.thunderforest.com/outdoors\" rel=\"noopener noreferrer\" target=\"_blank\">Gravitystorm</a> - Data &copy; <a href=\"http://www.openstreetmap.org/copyright\" rel=\"noopener noreferrer\" target=\"_blank\">OpenStreetMap</a>",
        "urlTemplate": "http://{s}.tile2.opencyclemap.org/transport/{z}/{x}/{y}.png",
        "minZoom": 0,
        "maxZoom": 20
    },
    {
        "id": "hot",
        "name": "Humanitarian (HOT)",
        "attribution": "Tiles &copy; <a href=\"http://hotosm.org\" rel=\"noopener noreferrer\" target=\"_blank\">HOT</a> - Data &copy; <a href=\"http://www.openstreetmap.org/copyright\" rel=\"noopener noreferrer\" target=\"_blank\">OpenStreetMap</a>",
        "urlTemplate": "https://{s}.tile.openstreetmap.fr/hot/{z}/{x}/{y}.png",
        "minZoom": 0,
        "maxZoom": 20
    },
    {
        "id": "openCycleMap",
        "name": "OpenCycleMap",
        "attribution": "Tiles &copy; <a href=\"http://www.opencyclemap.org\" rel=\"noopener noreferrer\" target=\"_blank\">OpenCycleMap</a> - Data &copy; <a href=\"http://www.openstreetmap.org/copyright\" rel=\"noopener noreferrer\" target=\"_blank\">OpenStreetMap</a>",
        "urlTemplate": "http://{s}.tile.opencyclemap.org/cycle/{z}/{x}/{y}.png",
        "minZoom": 0,
        "maxZoom": 20
    },
    {
        "id": "hikeBikeMap",
        "name": "HikeBikeMap",
        "attribution": "Tiles &copy; <a href=\"http://www.hikebikemap.org\" rel=\"noopener noreferrer\" target=\"_blank\">HikeBikeMap</a> - Data &copy; <a href=\"http://www.openstreetmap.org/copyright\" rel=\"noopener noreferrer\" target=\"_blank\">OpenStreetMap</a>",
        "urlTemplate": "http://{s}.tiles.wmflabs.org/hikebike/{z}/{x}/{y}.png",
        "minZoom": 0,
        "maxZoom": 20
    },
    {
        "id": "hydda",
        "name": "Hydda",
        "attribution": "Tiles &copy; <a href=\"http://openstreetmap.se\" rel=\"noopener noreferrer\" target=\"_blank\">OpenStreetMap Sweden</a> - Data &copy; <a href=\"http://www.openstreetmap.org/copyright\" rel=\"noopener noreferrer\" target=\"_blank\">OpenStreetMap</a>",
        "urlTemplate": "https://{s}.tile.openstreetmap.se/hydda/full/{z}/{x}/{y}.png",
        "minZoom": 0,
        "maxZoom": 18
    },
    {
        "id": "openTopoMap",
        "name": "OpenTopoMap",
        "attribution": "Tiles &copy; <a href=\"https://opentopomap.org\" rel=\"noopener noreferrer\" target=\"_blank\">OpenTopoMap</a> under <a href=\"https://creativecommons.org/licenses/by-sa/3.0\" rel=\"noopener noreferrer\" target=\"_blank\">CC BY SA 3.0</a> - Data &copy; <a href=\"http://www.openstreetmap.org/copyright\" rel=\"noopener noreferrer\" target=\"_blank\">OpenStreetMap</a>",
        "urlTemplate": "https://{s}.tile.opentopomap.org/{z}/{x}/{y}.png",
        "minZoom": 1,
        "maxZoom": 15
    },
    {
        "id": "openRiverboatMap",
        "name": "OpenRiverboatMap",
        "attribution": "Tiles &copy; <a href=\"http://fluv.io\" rel=\"noopener noreferrer\" target=\"_blank\">OpenRiverBoatMap</a> - Data &copy; <a href=\"http://www.openstreetmap.org/copyright\" rel=\"noopener noreferrer\" target=\"_blank\">OpenStreetMap</a>",
        "urlTemplate": "https://{s}.tile.openstreetmap.fr/openriverboatmap/{z}/{x}/{y}.png",
        "minZoom": 0,
        "maxZoom": 20
    },
    {
        "id": "osmFr",
        "name": "OpenStreetMap Français",
        "attribution": "Données &copy; <a href=\"http://www.openstreetmap.org/copyright\" rel=\"noopener noreferrer\" target=\"_blank\">OpenStreetMap</a>",
        "urlTemplate": "https://{s}.tile.openstreetmap.fr/osmfr/{z}/{x}/{y}.png",
        "minZoom": 0,
        "maxZoom": 20
    },
    {
        "id": "bano",
        "name": "BANO",
        "attribution": "<a href=\"https://openstreetmap.fr/bano\" rel=\"noopener noreferrer\" target=\"_blank\">BANO</a>",
        "urlTemplate": "http://{s}.layers.openstreetmap.fr/bano/{z}/{x}/{y}.png",
        "minZoom": 5,
        "maxZoom": 15
    },
    {
        "id": "cadastre",
        "name": "Cadastre Français",
        "attribution": "Données &copy; <a href=\"http://www.openstreetmap.org/copyright\" rel=\"noopener noreferrer\" target=\"_blank\">OpenStreetMap</a>",
        "urlTemplate": "http://tms.cadastre.openstreetmap.fr/*/tout/{z}/{x}/{y}.png",
        "minZoom": 0,
        "maxZoom": 22
    },
    {
        "id": "osmDe",
        "name": "OpenStreetMap Deutschland",
        "attribution": "Karte hergestellt aus <a href=\"http://osm.org\" rel=\"noopener noreferrer\" target=\"_blank\">OpenStreetMap-Daten</a> | Lizenz: <a href=\"http://opendatacommons.org/licenses/odbl\" rel=\"noopener noreferrer\" target=\"_blank\">Open Database License (ODbL)</a> | Courtesy of <a href=\"http://openstreetmap.de\" rel=\"noopener noreferrer\" target=\"_blank\">OpenStreetMap.de</a>",
        "urlTemplate": "http://{s}.tile.openstreetmap.de/tiles/osmde/{z}/{x}/{y}.png",
        "minZoom": 0,
        "maxZoom": 18
    }
]
