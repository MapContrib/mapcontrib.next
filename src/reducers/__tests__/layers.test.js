import {
  initialState,
  default as reducer,
  LayerNotFoundException
} from 'reducers/layers';
import {
  addLayer,
  addSourceToLayerById,
  removeLayerById
} from 'actions/layers';
import { OVERPASS_SOURCE } from 'const/layerSource';
import { CONTENT_DISPLAY_SIDEBAR } from 'const/contentDisplay';

const defaultLayerAttributes = {
  id: 'id',
  name: 'name',
  description: 'description',
  visible: true,
  leafletLayer: 'leafletLayer'
};

const overpassLayer = {
  ...defaultLayerAttributes,
  id: 'UUID_OVERPASS'
};

const osmoseLayer = {
  ...defaultLayerAttributes,
  id: 'UUID_OSMOSE'
};

const overpassSource = {
  id: 'id',
  type: OVERPASS_SOURCE,
  name: 'name',
  description: 'description',
  visible: true,
  contentDisplay: CONTENT_DISPLAY_SIDEBAR,
  content: 'content',
  leafletLayer: 'leafletLayer'
};

describe('Layers reducer', () => {
  it('Should return the initial state', () => {
    expect(reducer()).toBe(initialState);
  });

  it('Should be able to add a layer', () => {
    const state = reducer(initialState, addLayer(overpassLayer));

    expect(state).toMatchSnapshot();

    const state2 = reducer(state, addLayer(osmoseLayer));

    expect(state2).toMatchSnapshot();
    expect(state2).not.toBe(state);
  });

  it('Should be able to remove a layer', () => {
    const state = reducer(initialState, addLayer(overpassLayer));
    const state2 = reducer(state, addLayer(osmoseLayer));

    expect(Object.keys(state2).length).toBe(2);

    const state3 = reducer(state2, removeLayerById(overpassLayer.id));

    expect(Object.keys(state3).length).toBe(1);
    expect(state3).toMatchSnapshot();
    expect(state3).not.toBe(state2);
  });

  it('Should be able to add a source to a layer', () => {
    const state = reducer(initialState, addLayer(overpassLayer));
    const state2 = reducer(
      state,
      addSourceToLayerById(overpassLayer.id, overpassSource)
    );

    expect(state2).toMatchSnapshot();
    expect(state2).not.toBe(state);
  });

  it('Should throw an exception when adding a source to an unknown layer', () => {
    const state = reducer(initialState, addLayer(overpassLayer));
    const addSource = () =>
      reducer(state, addSourceToLayerById('Unknown_ID', overpassSource));

    expect(addSource).toThrow(LayerNotFoundException);
  });
});
