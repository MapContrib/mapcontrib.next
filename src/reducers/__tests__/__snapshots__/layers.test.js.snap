// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layers reducer Should be able to add a layer 1`] = `
Object {
  "UUID_OVERPASS": Object {
    "description": "description",
    "id": "UUID_OVERPASS",
    "isVisible": true,
    "name": "name",
    "representationType": "marker",
    "sources": Array [],
  },
}
`;

exports[`Layers reducer Should be able to add a layer 2`] = `
Object {
  "UUID_OSMOSE": Object {
    "description": "description",
    "id": "UUID_OSMOSE",
    "isVisible": true,
    "name": "name",
    "representationType": "marker",
    "sources": Array [],
  },
  "UUID_OVERPASS": Object {
    "description": "description",
    "id": "UUID_OVERPASS",
    "isVisible": true,
    "name": "name",
    "representationType": "marker",
    "sources": Array [],
  },
}
`;

exports[`Layers reducer Should be able to add a source to a layer 1`] = `
Object {
  "UUID_OVERPASS": Object {
    "description": "description",
    "id": "UUID_OVERPASS",
    "isVisible": true,
    "name": "name",
    "representationType": "marker",
    "sources": Array [
      "UUID_OVERPASS_SOURCE",
    ],
    "type": "OVERPASS",
  },
}
`;

exports[`Layers reducer Should be able to edit a layer 1`] = `
Object {
  "UUID_OVERPASS": Object {
    "description": "newDescription",
    "id": "UUID_OVERPASS",
    "isVisible": "newIsVisible",
    "name": "newName",
    "representationType": "newRepresentationType",
    "sources": Array [],
  },
}
`;

exports[`Layers reducer Should be able to remove a layer 1`] = `
Object {
  "UUID_OSMOSE": Object {
    "description": "description",
    "id": "UUID_OSMOSE",
    "isVisible": true,
    "name": "name",
    "representationType": "marker",
    "sources": Array [],
  },
}
`;
